bokehjs/src/coffee/common/hittest.coffee

```diff
@@ -35,7 +35,13 @@ class HitTestResult
       indices: []
     }
     # 2d for all for multilines and multi patches
-    @['2d'] = {indices: []}
+    @['2d'] = {
+      # index of the glyph in the multiglyph
+      indices: []
+      # object containing glyph_index: [point_index] of the nearest
+      # point to the crossed segment
+      point_indices: {}
+    }
```

Comment:
we should talk this over. My original intention was that for things like `MultiLine`:

* 1d indices would say "which line" 
* 2d indices would say "which segment" on the lines

Also instead of "nearest" I think for consistency we chose the first point of the segment. But thinking about it now, maybe this would work:
```
@['2d'] = {
    3 : [5],
    6 : [8, 20, 55]
}
```
that would say: lines 3 and 8 were hit, and the values give the segment points. I think we'd still want to fill in the "1d" attr as well (with 3 and 8)